v19.1/performance-best-practices-overview.md

```diff
@@ -57,24 +57,121 @@ This happens because when data is interleaved, queries that work on the parent t
 
 ## Unique ID best practices
 
-A traditional approach for generating unique IDs is one of the following:
+The best practices for generating unique IDs in a distributed database like CockroachDB are very different than for a legacy single-node database. Traditional approaches for generating unique IDs for legacy single-node databases include:
 
-- Monotonically increase `INT` IDs by using transactions with roundtrip `SELECT`s.
-- Use the [`SERIAL`](serial.html) pseudo-type for a column to generate random unique IDs.
+1. Using the [`SERIAL`](serial.html) pseudo-type for a column to generate random unique IDs. This can result in a performance bottleneck because IDs generated temporally near each other have similar values and are located physically near each other in a table's storage.
+2. Generating monotonically increasing [`INT`](int.html) IDs by using transactions with roundtrip [`SELECT`](select.html)s, e.g. `INSERT INTO tbl (id, …) VALUES ((SELECT max(id)+1 FROM tbl), …)`. This has a **very high performance cost** since it makes all [`INSERT`](insert.html) transactions wait for their turn to insert the next ID. You should only do this if your application really does require strict ID ordering. In some cases, using [Change Data Capture (CDC)](change-data-capture.html) can help avoid the requirement for strict ID ordering. If you can avoid the requirement for strict ID ordering, you can use one of the higher performance ID strategies outlined below.
 
-The first approach does not take advantage of the parallelization possible in a distributed database like CockroachDB.
+The approaches described above are likely to create hotspots for both reads and writes in CockroachDB. To avoid this issue, we recommend the following approaches (listed in order from best to worst performance). Each approach has pros and cons, however:
```

Comment:
```suggestion
The approaches described above are likely to create hotspots for both reads and writes in CockroachDB. To avoid this issue, we recommend the following approaches (listed in order from best to worst performance):
```

Reply:
Fixed.